broker/src/main/java/io/zeebe/broker/clustering/atomix/storage/snapshot/AtomixSnapshotStorage.java

```diff
@@ -142,31 +139,17 @@ public SnapshotMetrics getMetrics() {
   public void onNewSnapshot(
       final io.atomix.protocols.raft.storage.snapshot.Snapshot snapshot,
       final SnapshotStore store) {
-    final var snapshots = store.getSnapshots();
     metrics.incrementSnapshotCount();
     observeSnapshotSize(snapshot);
 
-    if (snapshots.size() >= maxSnapshotCount) {
-      // by the condition it's guaranteed there be a snapshot after skipping maxSnapshotCount - 1
-      @SuppressWarnings("squid:S3655")
-      final var oldest =
-          snapshots.stream()
-              .sorted(Comparator.reverseOrder())
-              .skip(maxSnapshotCount - 1L)
-              .findFirst()
-              .get();
-
-      LOGGER.info(
-          "Max snapshot count reached ({}), purging snapshots older than {}",
-          snapshots.size(),
-          oldest);
-      store.purgeSnapshots(oldest);
-
-      final var optionalConverted = toSnapshot(oldest.getPath());
-      if (optionalConverted.isPresent()) {
-        final var converted = optionalConverted.get();
-        deletionListeners.forEach(listener -> listener.onSnapshotsDeleted(converted));
-      }
+    LOGGER.info("Purging snapshots older than {}", snapshot);
```

Comment:
Probably debug level is enough here

Reply:
Where would we normally draw the line between info and debug?

Reply:
I think it is often not easy to decide. But maybe it helps to think from the user point of view. Does he needs to know it? What is the benefit or what can he do with this knowledge? Maybe also if we keep it on info the more context would be interesting. which dirs have been deleted

Reply:
Agreed. At this location it is not that interesting because purging does not necessarily mean that a snapshot was deleted. This is to be tackled in #4067. I've changed it to debug level now.